### DISCUSSION TITLE
weights normalisation in fit_optimize_positive function

### DISCUSSION THREAD

**Discussion by assuntaciarlo** [rsagroup/rsatoolbox#391] [Q&A]
Dear RSA community,

I was looking into the implementation of the fit_optimize_positive function which I understood to be suggested for weighted RSA models to obtain positive weights for the model components. I noticed that the function returns the  L2 normalised version of the estimated parameters whereas the loss function employed in the minimisation works on the unnormalised estimates. I was wondering if there was a specific reason for this choice instead of using the normalised parameters directly in the loss computation. And also do you think that this normalisation step is necessary if I want to compare weights across fittings on different data?

Many thanks in advance,
Assunta

**Comment by jdiedrichsen** [comment]
Hi all, 
I did raise this concern a while back, as for many applications you actually want the real coefficients (not normalized). 
@JasperVanDenBosch had some argument, I remember, but overall I think it would be much preferred to do this normalization outside of the function if required (or at least have an optional input parameter that allows you to get the non-normalized coefficients), 

Joern

**Reply by assuntaciarlo** [reply]
Thank you for the quick reply! I was indeed thinking that since I want to compare the weights from different fitting but both my data RDMs and model RDMs are defined using correlation distance, I might not need the normalisation. Is this correct? Furthermore, I was also wondering if the fitted RDM should be constrained to only contain values between 0 and 2 in case of using correlation distance. It should always be the case if the fitting is good but there might be some edge cases.
